src/client/localplayer.cpp

```diff
@@ -1096,6 +1103,22 @@ float LocalPlayer::getSlipFactor(Environment *env, const v3f &speedH)
 	return 1.0f;
 }
 
+float LocalPlayer::getSpeedFactor(Environment *env)
+{
+	const NodeDefManager *nodemgr = env->getGameDef()->ndef();
+	Map *map = &env->getMap();
+	const ContentFeatures &f = nodemgr->get(map->getNodeNoEx(
+			getStandingNodePos()));
+	int speed = 0;
+	if (f.walkable)
+		speed = itemgroup_get(f.groups, "speed");
+
+	if (speed != 0) {
+		return core::clamp(1.0f + (float)speed/10, 0.1f, 10.0f);
```

Comment:
I think dividing by 100 makes more sense here, to follow how `fall_damage_add_percent` works:

```suggestion
		return core::clamp(1.0f + (float)speed/100, 0.1f, 10.0f);
```

This means that -90 will decrease the player speed by 90%, and 100 would double it (increase it by 100%).

The documentation will have to be updated accordingly.

Reply:
Well, in MultiCraft I used %. But @stujones11 said it was "not in the spirit of the Minetest." What will others say?

Reply:
I'd prefer to use factors, rather than introducing percentages to the API.

Reply:
Well, there is an option to directly indicate the speed of the player instead of performing mathematically operations.

Reply:
I prefer division by 100 for more fine control, 10% steps are too big.

Reply:
I seem to recall that I did at one point suggest the 10% steps but I agree that division by 100 would be more intuitive and allow for finer grained control, sorry about the confusion

Reply:
@SmallJoker **speed = 0.5** or **speed = -50** for 1/2 on player speed?

Reply:
Groups are integer values so it has to be '-50' for half speed, we cannot use floats.